Api/Controllers/PicturesController.cs

```diff
@@ -0,0 +1,171 @@
+﻿using Domain.Services;
+using Domain.Services.Abstractions;
+using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.Mvc;
+
+namespace Api.Controllers
+{
+    [Route("api/pictures")]
+    [ApiController]
+    public class PicturesController : ControllerBase
+    {
+        private IPictureService pictureService;
+
+        public PicturesController(IPictureService pictureService)
+        {
+            this.pictureService = pictureService;
+        }
+
+        [HttpPost]
+        [Route("user/profile")]
+        [Produces("text/plain")]
+        [Authorize]
+        public ActionResult<string> PostUserProfilePicture(IFormFile picture)
+        {
+            try
+            {
+                using Stream s = picture.OpenReadStream();
+                string url = pictureService.AsUser(User.GetGuid()).UploadUserProfilePicture(s);
+                return Ok(url);
+            }
+            catch (UserUnauthorizedException e)
+            {
+                return Forbid(e.Message);
+            }
+            catch (StorageException e)
+            {
+                return this.FailedDependency(e.Message);
+            }
+            catch (Exception e)
+            {
+                return BadRequest(e.Message);
+            }
+        }
+
+        [HttpDelete]
+        [Route("user/profile")]
+        [Authorize]
+        public ActionResult DeleteUserProfilePicture()
+        {
+            try
+            {
+                pictureService.AsUser(User.GetGuid()).DeleteUserProfilePicture();
+                return Ok();
+            }
+            catch (UserUnauthorizedException e)
+            {
+                return Forbid(e.Message);
+            }
+            catch (StorageException e)
+            {
+                return this.FailedDependency(e.Message);
+            }
+            catch (Exception e)
+            {
+                return BadRequest(e.Message);
+            }
+        }
+
+        [HttpPost]
+        [Route("event/{target}")]
+        [Produces("text/plain")]
+        [Authorize]
+        public ActionResult<string> PostEventPicture([FromRoute] Guid target, [FromBody] IFormFile picture, [FromQuery] int index)
```

Comment:
I don't know if it is a problem on my side, but uploading picture through Swagger doesn't work for me, as it says that "The picture field is required". After removing [FromBody] it works.

Reply:
I know about this, wasted an hour trying to get it work on /user/profile endpoint, that's why there I removed [FromBody]. I simply forgot to remove it in the other two